impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -1567,21 +1567,34 @@ public int rank(final Set<ResourceType<?>> resourceTypes, final Collection<Servi
       Copier<K> keyCopier  = copyProvider.createKeyCopier(storeConfig.getKeyType(), storeConfig.getKeySerializer(), serviceConfigs);
       Copier<V> valueCopier = copyProvider.createValueCopier(storeConfig.getValueType(), storeConfig.getValueSerializer(), serviceConfigs);
 
+      List<Copier> copiers = new ArrayList<Copier>();
+      copiers.add(keyCopier);
+      copiers.add(valueCopier);
+
       SizeOfEngineProvider sizeOfEngineProvider = serviceProvider.getService(SizeOfEngineProvider.class);
       SizeOfEngine sizeOfEngine = sizeOfEngineProvider.createSizeOfEngine(
           storeConfig.getResourcePools().getPoolForResource(ResourceType.Core.HEAP).getUnit(), serviceConfigs);
       OnHeapStore<K, V> onHeapStore = new OnHeapStore<K, V>(storeConfig, timeSource, keyCopier, valueCopier, sizeOfEngine, eventDispatcher);
-      createdStores.add(onHeapStore);
+      createdStores.put(onHeapStore, copiers);
       return onHeapStore;
     }
 
     @Override
     public void releaseStore(Store<?, ?> resource) {
-      if (!createdStores.remove(resource)) {
+      List<Copier> copiers = null;
+      if ((copiers = createdStores.remove(resource)) == null) {
```

Comment:
Code readability matters. You declare the variable just the line before, why not assign it instead of having a more complicated construct in the if